framework/decode/custom_vulkan_ascii_consumer.h

```diff
@@ -50,7 +50,8 @@ inline std::string HandleIdToString(format::HandleId handleId)
     std::stringstream strStrm;
     if (handleId)
     {
-        strStrm << "\"0x" << reinterpret_cast<const void*>(handleId) << "\"";
+        const void* val = reinterpret_cast<const void*>(handleId);
```

Comment:
HandleId is typedef'd uint64_t, so will this `reinterpret_cast` truncate the handle on 32-bit?  Do you think it would work for us to just print the handleID? Like
`strStrm << "\"0x" << std::hex << handleId << "\"";`

Reply:
It works as well as the old code did on 32 bit. :-)
More seriously, do we support 32 bit, or have plans to? If so, this would be an issue.
I tried just printing the handleId as you suggested. The gcc compiler and VS compiler produce different results - one preceeds the output with a "0x", the other doesn't. I'm not sure why they produce different results. I tried to come up with the simplest code that produces the same output, this is what I found worked after lots of experiments.

Reply:
I dunno; maybe we shouldn't, but right now we do support 32-bit as far as I can tell.  :shrug:  It may be necessary for capturing some older apps that run in a translation layer.

The code
```
    uint64_t handleId = 0x50;
    strStrm << "\"0x" << std::hex << handleId << "\"";
```
results in `0x50` for me in Compiler Explorer for MSVC and gcc and clang, 32-bit and 64-bit.

Do you mean the pointer version may or may not print an "0x"?  For that one I suggested a cast to uintptr_t.

Reply:
When I tried this in the PtrToString function in the to_string.h change below, I get an extra '0x' - the original bug.  I couldn't get the gcc compiler not to prepend an 0x when outputting a ptr. So given the temp var and casting I had to do in the PtrToString, I decided to have HandleIdToString be the same. I can make them different if you want, but I would prefer they be similar.